provider/cmd/pulumi-gen-eks/main.go

```diff
@@ -1372,7 +1419,10 @@ func nodeGroupProperties(cluster, v2 bool) map[string]schema.PropertySpec {
 				"worker node.",
 		},
 		"nodeSecurityGroup": {
-			TypeSpec: schema.TypeSpec{Ref: awsRef("#/resources/aws:ec2%2FsecurityGroup:SecurityGroup")},
+			TypeSpec: schema.TypeSpec{
+				Ref:   awsRef("#/resources/aws:ec2%2FsecurityGroup:SecurityGroup"),
+				Plain: true,
+			},
```

Comment:
This is plain in the implementation:

https://github.com/pulumi/pulumi-eks/blob/c6f1fce98fa2683abe7ffbad7f166da99928fb83/nodejs/eks/nodegroup.ts#L66-L77